src/intelligent-placer.py

```diff
@@ -0,0 +1,46 @@
+from typing import Tuple, Optional
+
+import numpy as np
+import cv2
+
+
+def find_paper_and_polygon_contours(path_to_img: str) -> Tuple[Optional[np.ndarray], Optional[np.ndarray]]:
+    img = cv2.imread(path_to_img)
+    img_hsv = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
+
+    filter = cv2.inRange(img_hsv, np.array([0, 0, 169]), np.array([255, 255, 255]))
+    st = cv2.getStructuringElement(cv2.MORPH_RECT, (5, 5), (-1, -1))
+    filter = cv2.morphologyEx(filter, cv2.MORPH_CLOSE, st)
+    st = cv2.getStructuringElement(cv2.MORPH_RECT, (50, 50), (-1, -1))
+    filter = cv2.morphologyEx(filter, cv2.MORPH_OPEN, st)
+
+    contours, hierarchy = cv2.findContours(filter, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+
+    good_contours = []
+    cx = []
+    cy = []
+    polygon_contours = None
+    area_contours = []
+
+    min_contour_area = 10_000
+    prox_measure = 14
```

Comment:
Можно хранить только пару из макс площади и её индекса.
```suggestion
    prox_measure = 14
    
    max_area = 0
    max_area_idx = -1
    
    for i, cnt in enumerate(contours):
        contour_area = cv2.contourArea(cnt)
        if contour_area > max_area:
            max_area = contour_area
            max_area_idx = i
            
        if contour_area > min_contour_area:
            good_contours.append(cnt)
            M = cv2.moments(cnt)
            cx.append(int(M['m10'] / (M['m00'] + 1e-5)))
            cy.append(int(M['m01'] / (M['m00'] + 1e-5)))
    
```

Reply:
Действительно хорошее замечание. Исправил в новой версии